temporalio/client.py

```diff
@@ -4417,6 +4776,108 @@ async def terminate_workflow(self, input: TerminateWorkflowInput) -> None:
             req, retry=True, metadata=input.rpc_metadata, timeout=input.rpc_timeout
         )
 
+    async def start_workflow_update(
+        self, input: UpdateWorkflowInput
+    ) -> WorkflowUpdateHandle:
+        wait_policy = (
+            temporalio.api.update.v1.WaitPolicy(lifecycle_stage=input.wait_for_stage)
+            if input.wait_for_stage is not None
+            else None
+        )
+        req = temporalio.api.workflowservice.v1.UpdateWorkflowExecutionRequest(
+            namespace=self._client.namespace,
+            workflow_execution=temporalio.api.common.v1.WorkflowExecution(
+                workflow_id=input.workflow_id,
+                run_id=input.run_id or "",
+            ),
+            request=temporalio.api.update.v1.Request(
+                meta=temporalio.api.update.v1.Meta(
+                    update_id=input.update_id,
+                    identity=self._client.identity,
+                ),
+                input=temporalio.api.update.v1.Input(
+                    name=input.update,
+                ),
+            ),
+            wait_policy=wait_policy,
+        )
+        if input.args:
+            req.request.input.args.payloads.extend(
+                await self._client.data_converter.encode(input.args)
+            )
+        if input.headers is not None:
+            temporalio.common._apply_headers(
+                input.headers, req.request.input.header.fields
+            )
+        try:
+            resp = await self._client.workflow_service.update_workflow_execution(
+                req, retry=True, metadata=input.rpc_metadata, timeout=input.rpc_timeout
+            )
+        except RPCError as err:
+            # If the status is INVALID_ARGUMENT, we can assume it's an update
+            # failed error
+            if err.status == RPCStatusCode.INVALID_ARGUMENT:
+                raise WorkflowUpdateFailedError(input.workflow_id, input.update, err)
```

Comment:
Unlike queries, the update failure is in the success body, so you don't have to wrap certain RPC errors, they can all just be re-raised

Reply:
Ah, yeah good point